app-typescript/components/DatePicker.tsx

```diff
@@ -145,71 +131,83 @@ export class DatePicker extends React.PureComponent<IDatePicker, IState> {
             };
         }
 
+        if (this.props.preview) {
+            return (
+                // We have to do type assertion here because the core issue lies in primereact's calendar interface.
```

Comment:
It's not exactly that - primereact's interface is correct. The problem occurs because we wrap primereact's component using a narrower interface i.e. primereact supports an array of dates or a single date, and our wrapped component will only ever use a single date.